json2xml/index.js

```diff
@@ -0,0 +1,194 @@
+'use strict';
+/**
+ * This plugin accumulates data chunks from the client into an array
+ * property on the request or response, concats them on end. Then  
+ * transforms the request or response based on 'accept' or 'content-type' 
+ * headers. 
+ * 
+ * If the HTTP verb is GET and the Accept type does not match the 
+ * response type, the plugin will attempt to transform it (from 
+ * xml to json or vice versa).
+ *
+ * If the HTTP verb is NOT GET then the content-type is used. If the
+ * content-type is set to 'json', then an attempt is made to transform to
+ * 'xml' and vice-versa.
+ *
+ * This plugin can be disabled by sending the 'x-apigee-json2xml' http
+ * header.
+ *
+ * Users should be aware that buffering large requests or responses in
+ * memory can cause Apigee Edge Microgateway to run out of memory under
+ * high load or with a large number of concurrent requests. So this plugin
+ * should only be used when it is known that request/response bodies are small.
+ */
+
+var debug = require('debug')('plugin:json2xml');
+//library to convert json to xml
+var js2xmlparser = require("js2xmlparser");
+//library to convert xml to json
+var parseString = require("xml2js").parseString;
+var util = require("util");
+
+module.exports.init = function (config, logger, stats) {
+
+	//initialize the variables to false
+
+	//variables to control whether request and/or response transformation should take place
+	var requestXML = false;
+	var requestJSON = false;
+	var responseJSON = false;
+	var responseXML = false;
+	//use these variables to determine the content type of response sent by target server
+	var responseIsXML = false;
+	var responseIsJSON = false;
+	//flag to control whether transfornation should take place
+	var disable = false;
+
+	//method to accumulate responses
+	function accumulateResponse(res, data) {
```

Comment:
The accumulate plugin does not store the content anywhere so it can be accessed by a subsequent plugin. I'm ok if we were to change the accumulate plugin's behaviour and remove this code from here.